research/cifar10/ditto_mkmmd/client.py

```diff
@@ -0,0 +1,232 @@
+import argparse
+import os
+from logging import INFO
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from fl4health.checkpointing.checkpointer import BestLossTorchCheckpointer
+from fl4health.checkpointing.client_module import ClientCheckpointModule
+from fl4health.clients.mkmmd_clients.ditto_mkmmd_client import DittoMkmmdClient
+from fl4health.utils.load_data import load_cifar10_data, load_cifar10_test_data
+from fl4health.utils.losses import LossMeterType
+from fl4health.utils.metrics import Accuracy, Metric
+from fl4health.utils.random import set_all_random_seeds
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+from research.cifar10.model import ConvNet2
+
+NUM_CLIENTS = 10
+BASELINE_LAYERS = ["bn1", "bn2", "fc1"]
+
+
+class CifarDittoClient(DittoMkmmdClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        client_number: int,
+        learning_rate: float,
+        lam: float,
+        heterogeneity_level: float,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        mkmmd_loss_weight: float = 10,
+        feature_l2_norm_weight: float = 1,
+        mkmmd_loss_depth: int = 1,
+        beta_global_update_interval: int = 20,
+        checkpointer: Optional[ClientCheckpointModule] = None,
+    ) -> None:
+        super().__init__(
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            checkpointer=checkpointer,
+            lam=lam,
+            mkmmd_loss_weight=mkmmd_loss_weight,
+            flatten_feature_extraction_layers={key: True for key in BASELINE_LAYERS[-1 * mkmmd_loss_depth :]},
```

Comment:
We need this bools for feature extractor component. I am confused what do you mean by automatic, should we just change feature extractor component to flatten by default? because it is not doing it by default now.

Reply:
Oh now I see what you are saying. Yes sure we can do it.

Reply:
I will do this separately in mkmmd and deep mmd PRs.